python/cuml/feature_extraction/text.py

```diff
@@ -0,0 +1,603 @@
+        self.vocabulary_ = vocab[mask].reset_index(drop=True)
+
+        return count_df
+
+    def _preprocess(self, raw_documents):
+        preprocess = self.build_preprocessor()
+        return preprocess(raw_documents)
+
+    def fit(self, raw_documents):
+        """Build a vocabulary of all tokens in the raw documents.
+
+       Parameters
+       ----------
+       raw_documents : cudf.Series
+           A Series of string documents
+
+       Returns
+       -------
+       self
+       """
+        self.fit_transform(raw_documents)
+        return self
+
+    def fit_transform(self, raw_documents):
+        """Build the vocabulary and return document-term matrix.
+
+        Equivalent to .fit(X).transform(X) but preprocess X only once.
+
+        Parameters
+        ----------
+        raw_documents : cudf.Series
+           A Series of string documents
+
+        Returns
+        -------
+        X : array of shape (n_samples, n_features)
+            Document-term matrix.
+        """
+        self._warn_for_unused_params()
+        self._validate_params()
+
+        self._fixed_vocabulary = self.vocabulary is not None
+
+        docs = self._preprocess(raw_documents)
+        n_doc = len(docs)
+
+        tokenized_df = self._create_tokenized_df(docs)
+
+        if self._fixed_vocabulary:
+            self.vocabulary_ = self.vocabulary
+        else:
+            self.vocabulary_ = tokenized_df["token"].unique()
+
+        count_df = self._count_vocab(tokenized_df)
+
+        if not self._fixed_vocabulary:
+            max_doc_count = (self.max_df
+                             if isinstance(self.max_df, numbers.Integral)
+                             else self.max_df * n_doc)
+            min_doc_count = (self.min_df
+                             if isinstance(self.min_df, numbers.Integral)
+                             else self.min_df * n_doc)
+            if max_doc_count < min_doc_count:
+                raise ValueError(
+                    "max_df corresponds to < documents than min_df")
+            count_df = self._limit_features(count_df, self.vocabulary_,
+                                            max_doc_count,
+                                            min_doc_count,
+                                            self.max_features)
+
+        empty_doc_ids = self._compute_empty_doc_ids(count_df, n_doc)
+
+        X = self._create_csr_matrix_from_count_df(count_df, empty_doc_ids)
+        if self.binary:
+            X.data.fill(1)
+        return X
+
+    def transform(self, raw_documents):
+        """Transform documents to document-term matrix.
+
+        Extract token counts out of raw text documents using the vocabulary
+        fitted with fit or the one provided to the constructor.
+
+        Parameters
+        ----------
+        raw_documents : cudf.Series
+           A Series of string documents
+
+        Returns
+        -------
+        X : array of shape (n_samples, n_features)
```

Comment:
Datatype returned is a cupy CSR matrix right?

Reply:
Indeed, fixing this.